engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala

```diff
@@ -33,39 +34,57 @@ class FlinkRestManager(
   override def getProcessStates(
       name: ProcessName
   )(implicit freshnessPolicy: DataFreshnessPolicy): Future[WithDataFreshnessStatus[List[StatusDetails]]] = {
-    val preparedName = modelData.namingStrategy.prepareName(name.value)
+    getProcessesStates()
```

Comment:
1. We have fetching of all scenario states on two levels: in `DeploymentService` and in `FlinkRestManager`. It is quite tricky because when you analyze `DeploymentService` code you see some `None` for `prefetchedStatusDetailsWithFreshness`, you think that sth is done differently but eventually you figure out that it doesn't matter because in the deepest level `getProcessesStates` is invoked for both cases
2. We use "process stateS" in two context: when we get statuses of all jobs for given scenario and when we get statuses of all jobs for all scenarios. We need to distinguish it because it is hard to figure out what's going on
3. Do we still need the cache on FlinkClient level and/or on DM level?

Reply:
1. In DeploymentService, I split it into 2 methods, dedicated to those 2 cases: `getProcessStateFetchingStatusFromManager` and `getProcessStateUsingPrefetchedStatus`, without Options in signatures
2. In Nu we use names:
- `getProcessState` for fetching `ProcessState`
- `getProcessStates` for fetching `List[StatusDetails]`

Method in DeploymentService renamed `getProcessesStates` -> `getAllProcessesStates`, like the one in `StateQueryForAllScenariosSupported`. Is there anything more to rename in his PR?

3.
- IMO we can't remove caching on DM level, because all DM's use it, and this improvement is limited to Flink and Periodic.
- It may be possible to now remove cache on flink client level, but it should be IMO done outside this PR, separately, after testing the behavior without cache

Reply:
Ad. 1. Looks better
Ad. 2. No, it's more clear now
Ad. 3. Let's remove client level cache in some next PR